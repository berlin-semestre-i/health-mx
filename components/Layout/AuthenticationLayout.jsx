import React from 'react'
import { Grid, Container } from 'semantic-ui-react'
import styled from 'styled-components'
import media from 'styled-media-query'

const AuthenticationLayout = ({ children }) => (
  <AuthenticationGrid>
    <ImageColumn className="image-container" width={9}/>
    <FormColumn mobile={16} tablet={7} computer={7}>
      <FormContainer>
        { children }
      </FormContainer>
    </FormColumn>
  </AuthenticationGrid>
)

export default AuthenticationLayout

const AuthenticationGrid = styled(Grid)`
  && {
    height: 100vh;
    margin: 0;
    margin-left: -84px;
  }
`
const ImageColumn = styled(Grid.Column)`
  ${media.greaterThan('large')`
    &&&.image-container {
      padding: 0;
      background-image: url("../../static/images/enfermera.png");
      background-repeat: no-repeat;
      background-size: 100%;
    }
  `}

  ${media.lessThan('large')`
    &&&.image-container {
      display: none;
    }
  `}
`

const FormColumn = styled(Grid.Column)`
  & {
    margin: auto;
  }
`

const FormContainer = styled(Container)`
  ${media.lessThan('large')`
    & {
      padding: 0 1rem;
    }
  `}

  ${media.greaterThan('large')`
    & {
      padding: 0 4.5rem;
    }
  `}

  &&& .field, &&& .button {
    margin-bottom: 2em;
  }

  &&& .field label {
    font-size: 16px;
    font-weight: normal;
  }

  &&& .field label:after {
    display: none;
  }

  &&& img {
    max-width: 65%;
    margin-bottom: 3rem;
  }
`