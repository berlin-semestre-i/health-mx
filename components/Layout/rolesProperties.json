{
  "medic": {
    "placeholder": "Buscar derechohabiente o tarea",
    "background": "#3d5170"
  },
  "beneficiary": {
    "placeholder": "Buscar tarea o servicio",
    "background": "#0057A3"
  },
  "nurse": {
    "placeholder": "Buscar derechohabiente",
    "background": "#01493D"
  },
  "admin": {
    "placeholder": "Buscar usuario",
    "background": "#561A9B"
  },
  "error": {
    "placeholder": "Error: you have no role",
    "background": "#F65D61"
  }
}