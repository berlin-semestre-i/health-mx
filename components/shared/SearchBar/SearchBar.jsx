import React, { PureComponent } from 'react'
import { Input } from 'semantic-ui-react'
import styled from 'styled-components'

const Searchbar = styled(Input)`
  &&& {
    width: 100%;
  }

  &&&>input {
    border-radius: 0;
    height: 70px;
    border: none;
    box-shadow: 0 4px 10px -1px #e3e5e8;
    -webkit-box-shadow: 0 4px 10px -1px #e3e5e8;
  }
`

class SearchBar extends PureComponent {
  render() {
    return(
      <Searchbar icon="search" iconPosition="left" placeholder="Buscar derechohabiente o tarea"/>
    )
  }
}

export default SearchBar