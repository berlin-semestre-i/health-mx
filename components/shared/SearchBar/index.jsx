import React from 'react'
import { Input } from 'semantic-ui-react'
import styled from 'styled-components'

const SearchBar = ({ placeholder }) => (
  <Searchbar icon="search" iconPosition="left" placeholder={placeholder}/>
)

export default SearchBar

const Searchbar = styled(Input)`
  &&& {
    width: 100%;
  }

  &&&>input {
    font-size: 18px;
    font-weight: 300;
    border-radius: 0;
    height: 70px;
    border: none;
    box-shadow: 0 4px 10px -1px #e3e5e8;
    -webkit-box-shadow: 0 4px 10px -1px #e3e5e8;
  }
`
