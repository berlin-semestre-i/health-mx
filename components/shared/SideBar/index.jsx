import React from 'react'
import Link from 'next/link'
import { withRouter } from 'next/router'
import { Menu, Image, Icon, Popup } from 'semantic-ui-react'
import styled from 'styled-components'
import { getPropertiesFromRole } from '../../../utils/auth'
import { defaultManAvatar, defaultWomanAvatar } from '../../../utils/constants'

class SideBar extends React.PureComponent {
  render() {
    const { userRole, router, userAvatar, userGender } = this.props
    const roleProperties = getPropertiesFromRole(userRole)
    const genderDefaultAvatar = userGender === 'male' ? defaultManAvatar : defaultWomanAvatar
    const route = router.route

    return (
      <SideMenu
        name="sideMenu"
        vertical
        fixed="left"
        background={roleProperties.background}
        icon
      >
        <Link href={`/${userRole}`}>
          <HealthImage src="../../../static/images/Health-Bar-Logo.png"/>
        </Link>
        <Link href={`/${userRole}`}>
          <Menu.Item
            name="home"
            active={`/${userRole}` === route}
          >
            <Popup
              flowing
              inverted
              size="tiny"
              position="top right"
              horizontalOffset={12}
              trigger = {
                <Icon name="home" size="large"/>
              }
              content="Inicio"
            />
          </Menu.Item>
        </Link>
        { roleProperties.menuItems.map(item => (
          <Link href={`/${userRole}${item.address}`} key={item.id}>
            <Menu.Item
              name={item.name}
              active={`/${userRole}${item.address}` === route}
            >
              <Popup
                flowing
                inverted
                size="tiny"
                position="top right"
                horizontalOffset={12}
                trigger = {
                  <Icon name={item.icon} size="large"/>
                }
                content={item.description}
              />
            </Menu.Item>
          </Link>
        ))}
        <Notifications name="notifications">
          <Popup
            flowing
            inverted
            size="tiny"
            position="top right"
            horizontalOffset={12}
            trigger = {
              <Icon name="bell" size="large"/>
            }
            content="Notificaciones"
          />
        </Notifications>
        <Tips name="tips">
          <Popup
            flowing
            inverted
            size="tiny"
            position="top right"
            horizontalOffset={12}
            trigger = {
              <Icon name="question circle" size="large"/>
            }
            content="Ayuda"
          />
        </Tips>
        <Avatar
          name="userAvatar"
          src={userAvatar ? userAvatar : genderDefaultAvatar}
          circular
        />
      </SideMenu>
    )
  }
}

export default withRouter(SideBar)

const HealthImage = styled(Image)`
  && {
    margin: 15px 21px 36px 21px;
    width: 42px;
    height: 40px;
    cursor: pointer;
  }
`

const SideMenu = styled(Menu)`
  &&&& {
    display: flex;
    border: none;
    -webkit-box-shadow: 0 0 10px 1px #e7e9ed;
    box-shadow: 0 0 10px 1px #e7e9ed;
    background-color: ${props => props.background};
  }
`

const Avatar = styled(Image)`
  &&&& {
    background-color: #FFFFFF;
    width: 48.8px;
    height: 48.8px;
    margin: 35.6px 18px 50.6px 21px;
    align-self: flex-end;
    border: solid 2px #FFFFFF;
  }
`

const Notifications = styled(Menu.Item)`
  &&&& {
    margin-top: auto;
    align-self: flex-end;
  }
`

const Tips = styled(Menu.Item)`
  &&&& {
    margin-bottom: 0;
    align-self: flex-end;
  }
`