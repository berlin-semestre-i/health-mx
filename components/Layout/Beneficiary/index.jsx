import React, { Fragment } from 'react'
import SideBar from '../../shared/SideBar/SideBar'
import SearchBar from '../../shared/SearchBar/SearchBar'
import { CustomContainer, MainContainer } from '../../../static/styledComponents'

const Layout = ({ children }) => (
  <Fragment>
    <SideBar className="beneficiary"/>
    <CustomContainer>
      <SearchBar placeholder="Buscar tarea o servicio"/>
      <MainContainer>
        { children }
      </MainContainer>
    </CustomContainer>
  </Fragment>
)

export default Layout