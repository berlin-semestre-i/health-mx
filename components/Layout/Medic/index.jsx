import React from 'react'
import { Container } from 'semantic-ui-react'
import styled from 'styled-components'
import SideBar from '../../shared/SideBar/SideBar'
import SearchBar from '../../shared/SearchBar/SearchBar'

const CustomContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
  }
`

const MainContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
    padding: 3rem 3rem 0;
  }
`

const Layout = ({ children }) => (
  <React.Fragment>
    <SideBar background="#3d5170"/>
    <CustomContainer>
      <SearchBar />
      <MainContainer>
        { children }
      </MainContainer>
    </CustomContainer>
  </React.Fragment>
)

export default Layout