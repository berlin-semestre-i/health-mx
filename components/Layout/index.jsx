import React, { Fragment } from 'react'
import { Container } from 'semantic-ui-react'
import styled from 'styled-components'
import SideBar from '../shared/SideBar/SideBar'
import SearchBar from '../shared/SearchBar/SearchBar'
import { medic, beneficiary, nurse, error, admin } from './rolesProperties.json'

const getPropertiesFromRole = (userRole) => {
  if ( userRole === 'medic' ) {
    return medic
  }
  else if ( userRole === 'beneficiary' ) {
    return beneficiary
  }
  else if ( userRole === 'nurse' ) {
    return nurse
  }
  else if ( userRole === 'admin' ) {
    return admin
  }
  else {
    return error
  }
}

const Layout = ({ children, userRole }) => {
  const layoutProperties = getPropertiesFromRole(userRole)
  return (
    <Fragment>
      <SideBar background={layoutProperties.background}/>
      <GlobalContainer>
        <SearchBar placeholder={layoutProperties.placeholder}/>
        <MainContainer>
          { children }
        </MainContainer>
      </GlobalContainer>
    </Fragment>
  )
}

export default Layout

const GlobalContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
  }
`

const MainContainer = styled(Container)`
  &&& {
    margin: 0;
    width: 100%;
    padding: 3rem 3rem 0;
  }
`
