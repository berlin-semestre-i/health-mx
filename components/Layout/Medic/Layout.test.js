import React from 'react'
import { shallow, mount } from 'enzyme'
import Layout from './index'
import SearchBar from '../../shared/SearchBar/SearchBar'
import SideBar from '../../shared/SideBar/SideBar'

describe('Layout', () => {
  it('renders', () => {
    const component = shallow(<Layout />)
    expect(component).toMatchSnapshot()
  })


  it('should contain a SearchBar and a SideBar', () => {
    const wrapper = mount(<Layout />)
    expect(wrapper.contains(<SearchBar placeholder="Buscar derechohabiente o tarea"/>)).toEqual(true)
    expect(wrapper.contains(<SideBar className="standard"/>)).toEqual(true)
    wrapper.unmount()
  })
})