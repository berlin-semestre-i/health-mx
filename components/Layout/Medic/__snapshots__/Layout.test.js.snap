// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          width={1}
        >
          <Sidebar />
        </ForwardRef>,
        <ForwardRef
          width={15}
        >
          <ForwardRef>
            <SearchBar />
            <ForwardRef />
          </ForwardRef>
        </ForwardRef>,
      ],
      "columns": 2,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Sidebar />,
          "width": 1,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Medic__Column-sc-15h8q5n-2",
            "isStatic": true,
            "rules": Array [
              "&&&&{padding:0;}",
            ],
          },
          "create": [Function],
          "displayName": "Medic__Column",
          "handledProps": Array [
            "as",
            "children",
            "className",
            "color",
            "computer",
            "floated",
            "largeScreen",
            "mobile",
            "only",
            "stretched",
            "tablet",
            "textAlign",
            "verticalAlign",
            "widescreen",
            "width",
          ],
          "render": [Function],
          "styledComponentId": "Medic__Column-sc-15h8q5n-2",
          "target": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            <SearchBar />
            <ForwardRef />
          </ForwardRef>,
          "width": 15,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <SearchBar />,
              <ForwardRef />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "Medic__MainContainer-sc-15h8q5n-3",
                  "isStatic": true,
                  "rules": Array [
                    "&&&{margin:0;width:100%;padding:3rem 3rem 0;}",
                  ],
                },
                "displayName": "Medic__MainContainer",
                "handledProps": Array [
                  "as",
                  "children",
                  "className",
                  "content",
                  "fluid",
                  "text",
                  "textAlign",
                ],
                "render": [Function],
                "styledComponentId": "Medic__MainContainer-sc-15h8q5n-3",
                "target": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "Medic__CustomContainer-sc-15h8q5n-0",
              "isStatic": true,
              "rules": Array [
                "&&&{margin:0;width:100%;background-color:#f4f6f8;height:100vh;}",
              ],
            },
            "displayName": "Medic__CustomContainer",
            "handledProps": Array [
              "as",
              "children",
              "className",
              "content",
              "fluid",
              "text",
              "textAlign",
            ],
            "render": [Function],
            "styledComponentId": "Medic__CustomContainer-sc-15h8q5n-0",
            "target": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Medic__Column-sc-15h8q5n-2",
            "isStatic": true,
            "rules": Array [
              "&&&&{padding:0;}",
            ],
          },
          "create": [Function],
          "displayName": "Medic__Column",
          "handledProps": Array [
            "as",
            "children",
            "className",
            "color",
            "computer",
            "floated",
            "largeScreen",
            "mobile",
            "only",
            "stretched",
            "tablet",
            "textAlign",
            "verticalAlign",
            "widescreen",
            "width",
          ],
          "render": [Function],
          "styledComponentId": "Medic__Column-sc-15h8q5n-2",
          "target": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "Column": [Function],
      "Row": [Function],
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Medic__CustomGrid-sc-15h8q5n-1",
        "isStatic": true,
        "rules": Array [
          "&&&{margin:0;}",
        ],
      },
      "displayName": "Medic__CustomGrid",
      "handledProps": Array [
        "as",
        "celled",
        "centered",
        "children",
        "className",
        "columns",
        "container",
        "divided",
        "doubling",
        "inverted",
        "padded",
        "relaxed",
        "reversed",
        "stackable",
        "stretched",
        "textAlign",
        "verticalAlign",
      ],
      "render": [Function],
      "styledComponentId": "Medic__CustomGrid-sc-15h8q5n-1",
      "target": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            width={1}
          >
            <Sidebar />
          </ForwardRef>,
          <ForwardRef
            width={15}
          >
            <ForwardRef>
              <SearchBar />
              <ForwardRef />
            </ForwardRef>
          </ForwardRef>,
        ],
        "columns": 2,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Sidebar />,
            "width": 1,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "Medic__Column-sc-15h8q5n-2",
              "isStatic": true,
              "rules": Array [
                "&&&&{padding:0;}",
              ],
            },
            "create": [Function],
            "displayName": "Medic__Column",
            "handledProps": Array [
              "as",
              "children",
              "className",
              "color",
              "computer",
              "floated",
              "largeScreen",
              "mobile",
              "only",
              "stretched",
              "tablet",
              "textAlign",
              "verticalAlign",
              "widescreen",
              "width",
            ],
            "render": [Function],
            "styledComponentId": "Medic__Column-sc-15h8q5n-2",
            "target": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef>
              <SearchBar />
              <ForwardRef />
            </ForwardRef>,
            "width": 15,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <SearchBar />,
                <ForwardRef />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": undefined,
                  "componentStyle": ComponentStyle {
                    "componentId": "Medic__MainContainer-sc-15h8q5n-3",
                    "isStatic": true,
                    "rules": Array [
                      "&&&{margin:0;width:100%;padding:3rem 3rem 0;}",
                    ],
                  },
                  "displayName": "Medic__MainContainer",
                  "handledProps": Array [
                    "as",
                    "children",
                    "className",
                    "content",
                    "fluid",
                    "text",
                    "textAlign",
                  ],
                  "render": [Function],
                  "styledComponentId": "Medic__MainContainer-sc-15h8q5n-3",
                  "target": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "Medic__CustomContainer-sc-15h8q5n-0",
                "isStatic": true,
                "rules": Array [
                  "&&&{margin:0;width:100%;background-color:#f4f6f8;height:100vh;}",
                ],
              },
              "displayName": "Medic__CustomContainer",
              "handledProps": Array [
                "as",
                "children",
                "className",
                "content",
                "fluid",
                "text",
                "textAlign",
              ],
              "render": [Function],
              "styledComponentId": "Medic__CustomContainer-sc-15h8q5n-0",
              "target": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "Medic__Column-sc-15h8q5n-2",
              "isStatic": true,
              "rules": Array [
                "&&&&{padding:0;}",
              ],
            },
            "create": [Function],
            "displayName": "Medic__Column",
            "handledProps": Array [
              "as",
              "children",
              "className",
              "color",
              "computer",
              "floated",
              "largeScreen",
              "mobile",
              "only",
              "stretched",
              "tablet",
              "textAlign",
              "verticalAlign",
              "widescreen",
              "width",
            ],
            "render": [Function],
            "styledComponentId": "Medic__Column-sc-15h8q5n-2",
            "target": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Column": [Function],
        "Row": [Function],
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "Medic__CustomGrid-sc-15h8q5n-1",
          "isStatic": true,
          "rules": Array [
            "&&&{margin:0;}",
          ],
        },
        "displayName": "Medic__CustomGrid",
        "handledProps": Array [
          "as",
          "celled",
          "centered",
          "children",
          "className",
          "columns",
          "container",
          "divided",
          "doubling",
          "inverted",
          "padded",
          "relaxed",
          "reversed",
          "stackable",
          "stretched",
          "textAlign",
          "verticalAlign",
        ],
        "render": [Function],
        "styledComponentId": "Medic__CustomGrid-sc-15h8q5n-1",
        "target": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
